src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -105,9 +109,22 @@ public CopyArtifact(String projectName, String parameters, BuildSelector selecto
         this(projectName, parameters, selector, filter, null, target, flatten, optional, fingerprintArtifacts);
     }
 
-    @DataBoundConstructor
+    @Deprecated
     public CopyArtifact(String projectName, String parameters, BuildSelector selector, String filter, String excludes, String target,
                         boolean flatten, boolean optional, boolean fingerprintArtifacts) {
+        this(projectName);
+        setParameters(parameters);
+        setFilter(filter);
+        setTarget(target);
+        setExcludes(excludes);
+        setSelector(selector);
+        setFlatten(flatten);
+        setOptional(optional);
+        setFingerprintArtifacts(fingerprintArtifacts);
+    }
+
+    @DataBoundConstructor
+    public CopyArtifact(String projectName) {
         // check the permissions only if we can
         StaplerRequest req = Stapler.getCurrentRequest();
         if (req!=null) {
```

Comment:
BTW this block is legacy code, predating `QueueItemAuthenticator`; I thought @ikedam had already cleaned the permission system in this plugin up, but perhaps there is still some junk left behind. At any rate, it should be a no-op for Workflow since this constructor is called during the build, when there is no active web request. (It can _also_ be called by the snippet generator, when there is, but then there will be no `AbstractProject` in the context—only a `[Workflow]Job`.